### PR TITLE
[Monitor] Improve docs regarding sovereign clouds

### PR DESCRIPTION
**Pull request by pvaneck**
Since non-public cloud usage is a common usage scenario with this SDK, I am adding some samples and snippets to showcase usage with non-public clouds.

### CODE REVIEW

File: sdk/monitor/azure-monitor-ingestion/samples/sample_authentication.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+
+def authenticate_sovereign_cloud():
+    # [START create_client_sovereign_cloud]
+    from azure.identity import AzureAuthorityHosts, DefaultAzureCredential
+    from azure.monitor.ingestion import LogsIngestionClient
+
+    credential = DefaultAzureCredential()
+    endpoint = "https://example.ingest.monitor.azure.us"
+    client = LogsIngestionClient(endpoint, credential, credential_scopes=["https://monitor.azure.us//.default"])
```
Comment (scottaddie) [inline sdk/monitor/azure-monitor-ingestion/samples/sample_authentication.py:38]: Isn't this the scope for the Azure public cloud?

File: sdk/monitor/azure-monitor-ingestion/samples/async_samples/sample_authentication_async.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+async def authenticate_sovereign_cloud():
+    # [START create_client_sovereign_cloud_async]
+    from azure.identity import AzureAuthorityHosts
+    from azure.identity.aio import DefaultAzureCredential
+    from azure.monitor.ingestion.aio import LogsIngestionClient
+
+    credential = DefaultAzureCredential()
+    endpoint = "https://example.ingest.monitor.azure.us"
+    client = LogsIngestionClient(endpoint, credential, credential_scopes=["https://monitor.azure.us//.default"])
```
Comment (scottaddie) [inline sdk/monitor/azure-monitor-ingestion/samples/async_samples/sample_authentication_async.py:41]: Isn't this the scope for the Azure public cloud?

File: sdk/monitor/azure-monitor-ingestion/samples/sample_authentication.py (hunk 1 above)
Comment (pvaneck) [reply sdk/monitor/azure-monitor-ingestion/samples/sample_authentication.py:38]: The public cloud has `.com` while the US Government cloud has the `.us`.

Ref: https://learn.microsoft.com/en-us/azure/azure-monitor/app/azure-ad-authentication?tabs=net#token-audience

**Review by kristapratico — APPROVED** [review]
Looks really good! Just wondering, do we already have tests that test against sovereign clouds?

**Comment by pvaneck** [comment]
> Looks really good! Just wondering, do we already have tests that test against sovereign clouds?

Not yet in the CI, however I have manually tested on sovereign clouds several times. One of my work items this semester is to validate sovereign cloud support for a new batch metrics client, so I'll definitely use that as the opportunity to setup US/China cloud testing in the weekly pipeline.

**Pull request merged**